pkg/chartutil/coalesce.go

```diff
@@ -39,15 +39,46 @@ func concatPrefix(a, b string) string {
 //
 //   - Values in a higher level chart always override values in a lower-level
 //     dependency chart
+//
 //   - Scalar values and arrays are replaced, maps are merged
+//
 //   - A chart has access to all of the variables for it, as well as all of
 //     the values destined for its dependencies.
+//
+//     Note that CoalesceValues automatically removes nil values
 func CoalesceValues(chrt *chart.Chart, vals map[string]interface{}) (Values, error) {
```

Comment:
I think it could actually - currently `TrimNilValues` doesn't return the err on line 96 which I think could be improved as well, if that was done the sigatures should be compatible.

Ill give this a go tomorrow and test it out

Reply:
So it works out as:

```
func CoalesceValues(chrt *chart.Chart, vals map[string]interface{}) (Values, error) {
	valsCopy, err := MergeValues(chrt, vals)
	if err != nil {
		return vals, err
	}
	return TrimNilValues(valsCopy)
}
```

to avoid passing the error down into `TrimNilValues` - I'll run against my larger charts which show the original error tomorrow morning